api/internal/localizer/localizer.go

```diff
@@ -46,38 +44,53 @@ func NewLocalizer(ldr *LocLoader, validator ifc.Validator, rFactory *resmap.Fact
 		return nil, errors.WrapPrefixf(err, "unable to create directory in localize destination")
 	}
 	return &Localizer{
-		fSys:      ldr.fSys,
-		validator: validator,
-		rFactory:  rFactory,
-		pLdr:      pLdr,
-		ldr:       ldr,
-		dst:       filesys.ConfirmedDir(dst),
+		fSys:     ldr.fSys,
+		ldr:      ldr,
+		dst:      filesys.ConfirmedDir(dst),
+		rFactory: rFactory,
 	}, nil
 }
 
-// Localize localizes the root that lc is at
 func (lc *Localizer) Localize() error {
-	kt := target.NewKustTarget(lc.ldr, lc.validator, lc.rFactory, lc.pLdr)
-	err := kt.Load()
+	err := lc.loadKust()
 	if err != nil {
-		return errors.Wrap(err)
+		return err
 	}
 
-	kust := lc.processKust(kt)
+	err = lc.processKust()
+	if err != nil {
+		return err
+	}
 
-	content, err := yaml.Marshal(kust)
+	locKust, err := lc.kust.Marshal()
 	if err != nil {
-		return errors.WrapPrefixf(err, "unable to serialize localized kustomization file")
+		return err
 	}
-	if err = lc.fSys.WriteFile(lc.dst.Join(konfig.DefaultKustomizationFileName()), content); err != nil {
-		return errors.WrapPrefixf(err, "unable to write localized kustomization file")
+	err = lc.fSys.WriteFile(lc.dst.Join(lc.kustFileName), locKust)
```

Comment:
`kustTarget` has a `kustFileName` that you can use directly, if we go with the approach I suggested above.

Reply:
Are you suggesting we make the `kustFileName` field on `kustTarget` public?